pkg/kubelet/kubelet.go

```diff
@@ -932,6 +952,87 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 	return node, nil
 }
 
+// retrieveNodeLabels applied the node labels specified to the
+func (kl *Kubelet) retrieveNodeLabels(labels []string) (map[string]string, error) {
+	var fileLabels []string
+	nodeLabels := make(map[string]string, 0)
+
+	// step: iterate the files first of all
+	for _, label := range labels {
+		if strings.HasPrefix(label, "file:///") {
+			// @TODO perhaps just change this to strings.TrimPrefix()
+			fp, err := url.Parse(label)
+			if err != nil {
+				return nodeLabels, fmt.Errorf("failed to parse the node label file path %s, %s", label, err)
+			}
+
+			tags, err := kl.readNodeLabelsFile(fp.Path)
+			if err != nil {
+				return nodeLabels, fmt.Errorf("failed to parse node label file %s, %s", fp.Path, err)
+			}
+
+			fileLabels = append(fileLabels, tags...)
+		} else {
+			fileLabels = append(fileLabels, label)
+		}
+	}
+
+	for _, label := range fileLabels {
+		items := strings.Split(label, "=")
+		if len(items) != 2 {
+			return nodeLabels, fmt.Errorf("invalid node label: %s, should be in the form key=pair", label)
+		}
+		// trim any whitespace
+		nodeLabels[strings.TrimSpace(items[0])] = strings.TrimSpace(items[1])
+	}
+
+	return nodeLabels, nil
+}
+
+// readNodeLabelsFile reads in and parses the node labels file
+func (kl *Kubelet) readNodeLabelsFile(path string) ([]string, error) {
+	var list []string
+
+	if found, err := util.FileExists(path); err != nil {
+		return nil, fmt.Errorf("failed to stat the node label file %s, %s", path, err)
+	} else if !found {
+		return nil, fmt.Errorf("the node label file %s does not exist", path, err)
+	}
+
+	fd, err := os.Open(path)
+	if err != nil {
+		return list, fmt.Errorf("failed to open the node labels file %s, %s", path, err)
+	}
+
+	defer fd.Close()
+	rdr := bufio.NewReader(fd)
+
+	for {
```

Comment:
I feel like this might be cleaner as:

```go
for line, prefix, err := rd.ReadLine(); err != io.EOF {
   if err != nil {
     return nil, err
   }
   ...
}